core/internal/api/ratelimitheaders.go

```diff
@@ -0,0 +1,38 @@
+package api
+
+import (
+	"net/http"
+	"strconv"
+)
+
+// Values of the RateLimit headers defined in
+// https://www.ietf.org/archive/id/draft-polli-ratelimit-headers-02.html
+type RateLimitHeaders struct {
+	// The RateLimit-Remaining header.
+	//
+	// Guaranteed non-negative.
```

Comment:
i'm not sure it is true, based on the doc. are we going to enforce it in the client?

Reply:
it's enforced in the parsing code below